corehq/apps/userreports/views.py

```diff
@@ -124,73 +133,108 @@ def create_report(request, domain):
     return _edit_report_shared(request, domain, ReportConfiguration(domain=domain))
 
 
-class ReportBuilderView(TemplateView):
+class ReportBuilderView(BaseDomainView):
 
     @cls_to_view_login_and_domain
-    @upgrade_knockout_js
+    @use_bootstrap3
+    @use_knockout_js
+    @use_select2
+    @use_daterangepicker
     @method_decorator(toggles.REPORT_BUILDER.required_decorator())
     @method_decorator(requires_privilege_raise404(privileges.REPORT_BUILDER))
-    def dispatch(self, request, domain, **kwargs):
-        self.domain = domain
-        return super(ReportBuilderView, self).dispatch(request, domain, **kwargs)
+    def dispatch(self, request, *args, **kwargs):
+        return super(ReportBuilderView, self).dispatch(request, *args, **kwargs)
 
+    @property
+    def section_name(self):
+        return _("Report Builder")
 
-class ReportTypeTileConfiguration(TileConfiguration):
-    def __init__(self, *args, **kwargs):
-        self.analytics_label = kwargs.pop('analytics_label', None)
-        super(ReportTypeTileConfiguration, self).__init__(*args, **kwargs)
+    @property
+    def section_url(self):
+        return reverse(ReportBuilderTypeSelect.urlname, args=[self.domain])
 
 
-class ReportBuilderTypeSelect(ReportBuilderView):
+class ReportBuilderTypeSelect(JSONResponseMixin, ReportBuilderView):
     template_name = "userreports/builder_report_type_select.html"
+    urlname = 'report_builder_select_type'
+    page_title = ugettext_noop('Select Report Type')
```

Comment:
Here it should either be `ugettext_lazy` or `ugettext_noop`, otherwise it'll translate before it has a request (and locale).  [Here's a description of the difference](https://commcare-hq.readthedocs.org/en/latest/translations.html#tagging-strings-in-views).

`ugettext_noop` doesn't translate at all, so you'd need to explicitly translate this again later: `ugettext(report.page_title)` (which might already be happening, not sure).

You can also use `ugettext_lazy`, then when you call `json.dumps`, you pass in the kwarg [`default=json_handler`](https://github.com/dimagi/dimagi-utils/blob/master/dimagi/utils/web.py#L179-L196) (from `dimagi.utils.web`).  That will force any lazy translations at that point.